File: Algorithms/src/1957. Delete Characters to Make Fancy String/Code/solution.cpp
class Solution
{
public:
    string makeFancyString(string s)
    {
        string result;
        for (char c : s)
        {
            int n = result.size();
            if (n < 2 || !(result[n - 1] == c && result[n - 2] == c))
            {
                result += c;
            }
        }
        return result;
    }
};

File: Algorithms/src/1957. Delete Characters to Make Fancy String/Code/solution.go
func makeFancyString(s string) string {
    result := []byte{}
    for i := 0; i < len(s); i++ {
        n := len(result)
        if n < 2 || !(result[n-1] == s[i] && result[n-2] == s[i]) {
            result = append(result, s[i])
        }
    }
    return string(result)
}